### PR TITLE
Contributing PepperBox - Enhanced kafka support plugin

### PR DESCRIPTION
**Pull request by Nachiket90**
This plugin provides below enhanced features to user as compared to existing kafkameter plugin. 
1. Create customized message/schema.
2. Implementation of popular security protocols like SSL, SASL etc.
3. High throughput (~0.5M/sec messages with 8 core CPU and 16 GB memory server)

### CODE REVIEW

File: site/dat/repo/various.json [hunk 1]
```diff
@@ -138,6 +138,38 @@
...
     }
   },
+  {
+    "id": "PepperBox",
+    "name": "Enhanced Kafka Support with customized schema and SASL support",
+    "description": "This plugin provides an easy way to generate load on Kafka with customized schema, high throughput and with popular security protocols",
+    "screenshotUrl": "",
+    "helpUrl": "https://github.com/GSLabDev/pepper-box",
+    "vendor": "www.gslab.com",
+    "markerClass": "com.gslab.pepper.PepperBoxLoadGenerator.java",
```
Comment (undera) [inline site/dat/repo/various.json]: `markerClass` has to name just fully qualified Java class name, not file name. Looks like `.java` suffix here and in `componentClasses` is not what you intend. Please review.

File: site/dat/repo/various.json [hunk 2]
```diff
@@ -138,6 +138,38 @@
...
+    "componentClasses": [
+	  "com.gslab.pepper.input.SchemaParser.java",
+	  "com.gslab.pepper.input.SchemaProcessor.java",
+	  "com.gslab.pepper.loadgen.impl.PlaintTextLoadGenerator.java",
+	  "com.gslab.pepper.loadgen.impl.SerializedLoadGenerator.java",
+	  "com.gslab.pepper.model.FieldExpressionMapping.java",
+	  "com.gslab.pepper.sampler.PepperBoxKafkaSampler.java"
+    ],
+    "versions": {
+      "v1.0": {
```
Comment (undera) [inline site/dat/repo/various.json]: Version names have to be just numbers, no `v` letter is allowed.

File: site/dat/repo/various.json [hunk 3]
```diff
@@ -138,6 +138,38 @@
...
+        "downloadUrl": "https://github.com/GSLabDev/pepper-box/archive/v1.0.zip",
```
Comment (undera) [inline site/dat/repo/various.json]: ZIP file won't be handled, `downloadUrl` has to point to `.jar` file download.

Comment (Nachiket90) [reply site/dat/repo/various.json]: I am done with changes but I see other plugins are providing jar downloadUrl as https://jmeter-plugins.org/files/thirdparty/. Can you share the procedure/guidelines to upload file there if any?

Comment (undera) [reply site/dat/repo/various.json]: /thirdparty/ is something I have built once to publish some abandoned plugins. You have to manage uploads yourself, but it's quite easy with github. You can upload your JAR into any tag, there is upload feature.
But you need to beware of one requirement: your JAR filename has to follow Maven's naming convention, where version name is part of JAR filename, plus restriction on no 'v' letter. 
For simplicity, I suggest you to work by analogy with this repo: https://github.com/Blazemeter/jmeter-bzm-plugins/releases

Comment (Nachiket90) [reply site/dat/repo/various.json]: Thanks for information. I'll make the suggested changes about jar filename.

**Comment by undera** [comment]
Looks dead

**Pull request closed without merging**